### PR TITLE
decompressor: fixing a bug for requests with added trailers

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: Medium: filter chain change but only (?) for added trailers
Testing: new integration test
Docs Changes: n/a
Release Notes: inline
Runtime guard: envoy.reloadable_features.fix_added_trailers

h/t to @rgs1 for the bug report, and writing the integration test!

Co-authored-by: @rgs1

### CODE REVIEW

**Review by rgs1 — APPROVED** [review]
Thanks! Adding it to our build now.

**Comment by rgs1** [comment]
Oops:

```
RSTChecker NOTICE [current_version] Running check
RSTChecker ERROR [current_version] (docs/root/version_history/current.rst:76) Backticks should come in pairs (except for links and refs): * compressor: fix a bug where if trailers were added and a subsequent filter paused the filter chain, the request could be stalled. This behavior can be reverted by setting `envoy.reloadable_features.fix_added_trailers` to false.
RSTChecker ERROR [current_version] (docs/root/version_history/current.rst:77) Version history not in alphabetical order (compressor vs aws request signer): please check placement of line

```

File: docs/root/version_history/current.rst [hunk 1]
```diff
@@ -73,6 +73,7 @@ Bug Fixes
 
 * access log: fix ``%UPSTREAM_CLUSTER%`` when used in http upstream access logs. Previously, it was always logging as an unset value.
 * access log: fix ``%UPSTREAM_CLUSTER%`` when used in http upstream access logs. Previously, it was always logging as an unset value.
+* compressor: fix a bug where if trailers were added and a subsequent filter paused the filter chain, the request could be stalled. This behavior can be reverted by setting `envoy.reloadable_features.fix_added_trailers` to false.
```
Comment (rgs1) [inline docs/root/version_history/current.rst:76]: sort and you need two back ticks these days (see UPSTREAM_CLUSTER above)

**Review by rgs1 — CHANGES_REQUESTED** [review]

**Review by mattklein123 — APPROVED** [review]
Sweet, thanks for the quick turnaround!

**Review by rgs1 — APPROVED** [review]

**Review by snowp — APPROVED** [review]
Thanks!

**Pull request merged**